client/src/unifyfs-sysio.c

```diff
@@ -1057,8 +1005,19 @@ ssize_t UNIFYFS_WRAP(write)(int fd, const void* buf, size_t count)
             return (ssize_t)(-1);
         }
 
+        if (filedesc->append) {
+            /*
+             * With O_APPEND we always write to the end, despite the current
+             * file position.
+             */
+            int fid = unifyfs_get_fid_from_fd(fd);
+            pos = unifyfs_fid_local_size(fid);
```

Comment:
Do we need to update filedesc->pos to point to the end of the file?

Reply:
We do a few lines down:
```C
      if (filedesc->append) {
            /*
             * With O_APPEND we always write to the end, despite the current
             * file position.
             */
            int fid = unifyfs_get_fid_from_fd(fd);
            pos = unifyfs_fid_local_size(fid);
        } else {
            pos = filedesc->pos;
        }

        /* write data to file */
        int write_rc = unifyfs_fd_write(fd, pos, buf, count);
        if (write_rc != UNIFYFS_SUCCESS) {
            errno = unifyfs_err_map_to_errno(write_rc);
            return (ssize_t)(-1);
        }
        ret = count;

        /* update file position */
        filedesc->pos += ret;
```

Reply:
Is that the correct value, though?  Can someone do something like this?
```
fd = open(path, O_APPEND) // consider that file exists and is already 200 bytes
lseek(fd, 100, SEEK_SET) // can this move filedesc->pos to 100?
write(fd, 1) // write one byte at the end (offset=200)
```
However then ```filedesc->pos += 1``` so that now ```filedesc->pos = 101```.

Should it be 201 instead?  I don't know.

Reply:
You're correct, for append it should be `filedesc->pos = filesize + count`:
```
O_APPEND The file is opened in append mode. Before each write(2), the file offset is positioned at the end of the file.
```

Reply:
Thanks @tonyhutter , one more subtle detail.

What happens in the case of a write error?  Based on the man page, it sounds like the position is updated, and then then write is attempted.  Does that mean we need to move the position to the end, regardless of whether the write is successful?  If so, I'm thinking we should do a

```
int fid = unifyfs_get_fid_from_fd(fd);
pos = unifyfs_fid_local_size(fid);
filedesc->pos = pos
```
then later, we could keep the line as:
```
filedesc->pos += ret;
```

Reply:
That make sense